lib/krb5/krb5.conf.5

```diff
@@ -518,6 +518,10 @@ If this flag is true, then all application protocol authentication
 requests will be flagged to indicate that the application supports
 channel bindings when operating over a secure channel.
 The default value is false.
+.It Li gss_canonicalize_source_name = Va boolean
```

Comment:
Why default to false?  Why not true?  What might break?  Why have this be configurable?  Why is this in GSS and not in the `rd_req` side in `lib/krb5`?  Wouldn't we want this to be used for non-GSS apps?  (Well, we don't want non-GSS krb5 apps, but that's another story.)

If this should move to `lib/krb5`, maybe rename this param to `use_client_name_from_pac`.

Reply:
Also, with the `Principal` decoration I'm adding, maybe we could stuff the canonical and alias names in there and provide an accessor (name attributes).

Reply:
> Why default to false? Why not true? What might break? Why have this be configurable? Why is this in GSS and not in the `rd_req` side in `lib/krb5`? Wouldn't we want this to be used for non-GSS apps? (Well, we don't want non-GSS krb5 apps, but that's another story.)
> 
> If this should move to `lib/krb5`, maybe rename this param to `use_client_name_from_pac`.

Can move to `lib/krb5`, good idea. Defaulting to TRUE will change behaviour where aliases are used without the `canonicalize` flag which would break @jaltman's use case.

Reply:
Also, note, because of the rd_req API, this will involve replacing `ticket->client` with the canonical name from the PAC; the ticket client will be lost forever.